### PR TITLE
Fabric 7: TypeScript 3.5 Updates (WIP)

### LINKED ISSUE #9038: Target TS3.5
Figure out when RCs are going to drop so that we can plan ahead

### PR DESCRIPTION
**Pull request by JasonGore**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9038
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

WIP TS3.5 updates on `fabric-7` branch.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9209)

### CODE REVIEW

File: packages/utilities/src/properties.ts [hunk 1]
```diff
@@ -277,6 +277,10 @@ export const imageProperties = divProperties;
  * @returns The filtered props
  */
 export function getNativeProps<T>(props: {}, allowedPropNames: string[], excludedPropNames?: string[]): T {
+  // It'd be great to properly type this while allowing 'aria-` and 'data-' attributes like TypeScript does for JSX attributes,
+  // but that ability is hardcoded into the TS compiler with no analog in TypeScript typings.
+  // Then we'd be able to enforce props extends native props (including aria- and data- attributes), and then return native props.
+  // We should be able to do this once this PR is merged: https://github.com/microsoft/TypeScript/pull/26797
```
Comment (cliffkoh) [inline packages/utilities/src/properties.ts]: :+1:

File: scripts/package.json [hunk 2]
```diff
@@ -46,8 +46,8 @@
     "jju": "^1.4.0",
     "jscodeshift": "^0.5.0",
     "json-loader": "^0.5.7",
-    "just-scripts": "~0.14.4",
-    "just-task": "~0.9.5",
+    "just-scripts": "0.14.4",
```
Comment (dzearing) [inline scripts/package.json]: intentional?

Comment (JasonGore) [reply scripts/package.json]: Yes, updating the shrinkwrap brought in a new version of Just that the repo wasn't ready for. I had to pin in for now until the changes Ken made get merged in. I'll be sure to revert before merge.

**Comment by JasonGore** [comment]
General Status: I have more things to fix in this PR, but OUFR is building so I think I'm over the hump. I wanted to disable "draft" status so CI would start running.

File: packages/example-app-base/src/components/MarkdownTable/MarkdownTable.types.ts [hunk 3]
```diff
@@ -15,7 +15,8 @@ export interface IMarkdownTableCellProps extends IMarkdownTableProps {
    * Render the table cell as a th or td.
    * @default 'td'
    */
-  as: 'th' | 'td';
+  // TODO: doc says default is 'td', where is it set? if it has a default, why is 'as' required? making optional for now.
+  as?: 'th' | 'td';
```
Comment (JasonGore) [inline packages/example-app-base/src/components/MarkdownTable/MarkdownTable.types.ts]: @ecraig12345 FYI. The watch issue combined with my lack of knowledge with this code made it hard to iterate and find a good fix for this.

File: packages/example-app-base/src/components/Page/Page.types.ts [hunk 4]
```diff
@@ -161,6 +161,11 @@ export interface IPageSectionProps<TPlatforms extends string = string>
 }
 
 /** Version of IPageSectionProps where `sectionName` is required. */
-export type IPageSectionPropsWithSectionName = Required<Pick<IPageSectionProps, 'sectionName'>> & Omit<IPageSectionProps, 'sectionName'>;
+// TODO: I'm not sure the best way to fix this, and the TS watch issue is making it harder to iterate and try fixes.
+//        Equating types with a slight loss in type safety for now.
+export type IPageSectionPropsWithSectionName = IPageSectionProps;
+// export type IPageSectionPropsWithSectionName<TPlatform extends string = string> =
+//   Required<Pick<IPageSectionProps<TPlatform>, 'sectionName'>> & Omit<IPageSectionProps<TPlatform>, 'sectionName'>;
+// export type IPageSectionPropsWithSectionName = Required<Pick<IPageSectionProps, 'sectionName'>> & Omit<IPageSectionProps, 'sectionName'>;
```
Comment (JasonGore) [inline packages/example-app-base/src/components/Page/Page.types.ts]: @ecraig12345 FYI. The watch issue combined with my lack of knowledge with this code made it hard to iterate and find a good fix for this.

File: packages/example-app-base/src/components/Page/sections/ExamplesSection.tsx [hunk 5]
```diff
@@ -5,7 +5,9 @@ import * as styles from '../Page.module.scss';
 
 export interface IExamplesSectionProps extends IPageSectionPropsWithSectionName {
   exampleKnobs?: React.ReactNode;
-  examples: IExample[];
+  // TODO: There seems to be a disparity between this type and IPageSectionProps as used in Page.tsx.
+  //        Making optional for now to workaround.
+  examples?: IExample[];
```
Comment (JasonGore) [inline packages/example-app-base/src/components/Page/sections/ExamplesSection.tsx]: @ecraig12345 One more...

**Comment by JasonGore** [comment]
CI (aside from CLA) appears not to be running against this PR.

**Comment by JasonGore** [comment]
I'm not sure why CI on this PR remains busted. Perhaps it's because it started as a draft PR? I'm going to close this PR and create another.

File: packages/experiments/src/components/PersonaCoin/PersonaCoinInitials/PersonaCoinInitials.tsx [hunk 6]
```diff
@@ -1,13 +1,19 @@
...
 import { Icon, Text } from 'office-ui-fabric-react';
+import { ISlotProp } from '../../../Foundation';
 import { IPersonaCoinProps } from '../PersonaCoin.types';
 import { getInitials, getRTL } from '../../../Utilities';
 
+export type IPersonaCoinInitialsSlot = ISlotProp<IPersonaCoinInitialsProps, string>;
+
+// TODO: Have Markus verify... className props was required, but I'm not sure if it should be.
+// If it is truly required, it seems the parent component should be ensuring that default values are
+// provided and not assume the consumer of the slot will provide them.
```
Comment (JasonGore) [inline packages/experiments/src/components/PersonaCoin/PersonaCoinInitials/PersonaCoinInitials.tsx]: @Markionium FYI, wasn't sure which way to go here. (Sorry for the typo on your name.)

**Comment by JasonGore** [comment]
@msft-github-bot Please merge in 1 minute

**Pull request merged**